todo_app/data/trello_items.py

```diff
@@ -0,0 +1,136 @@
+import requests
+import json
+from todo_app.data.item import Item
+from todo_app.data.trelloList import TrelloList
+import todo_app.data.trello_constants as constants
+
+class Trello_service(object):
+    trello_lists = {}
+    items = []
+
+    def __init__(self):
+        self.get_trello_secrets()
+        self.get_lists()
+        self.get_items_from_trello()
+
+    def get_trello_secrets(self):
+        trello_key = ''
+        trello_token = ''
+        trello_secrets = []
+        with open(constants.TRELLO_KEYS_PATH, 'r') as file:
+            split_lines = [line.split('=') for line in file.read().splitlines()]
+            trello_secrets = [{'k': k, 'v': v} for [k, v] in split_lines]
+        
+        for item in trello_secrets:
+            k = item['k']
+            v = item['v']
+            if k == "key":
+                trello_key = v
+            else:
+                trello_token = v
+        
+        constants.TRELLO_CREDENTIALS = f"key={trello_key}&token={trello_token}"
+
+    def get_lists(self):
+        """
+        Fetches all lists from Trello Api.
+
+        """
+        url = f"{constants.TRELLO_API_URL}boards/{constants.TRELLO_BOARD_ID}/lists?{constants.TRELLO_CREDENTIALS}"
+        response = requests.request("GET", url)
+        raw_lists = (json.loads(response.text.encode('utf8')))
+        for trello_list in raw_lists:
+            trelloListDict = TrelloList(name=trello_list[constants.TRELLO_NAME], 
+                                        boardId=trello_list[constants.TRELLO_ID_BOARD])
+            self.trello_lists[trello_list[constants.TRELLO_ID]] = trelloListDict
+
+    def get_list_id(self, name):
+        """
+        Get a trello list id for a give name from the list of all lists.
+
+        Returns:
+            listId:  The identifier for a given name
+        """
+        for listId in self.trello_lists:
+            trello_list = self.trello_lists[listId]
+            if trello_list.name == name:
+                return listId
+
+    def get_items_from_trello(self):
+        """
+        Fetches all saved items from Trello.
+
+        Returns:
+            list: The list of saved items.
+        """
+        self.items.clear()
+        url = f"{constants.TRELLO_API_URL}boards/{constants.TRELLO_BOARD_ID}/cards?{constants.TRELLO_CREDENTIALS}"
+        response = requests.request("GET", url)
+        cards = json.loads(response.text.encode('utf8'))
+        i = 0
+        for card in cards:
+            trelloListDict = self.trello_lists[card[constants.TRELLO_IDLIST]]
+            item = Item(id=i, 
+                        cardID=card[constants.TRELLO_ID], 
+                        status=trelloListDict.name, 
+                        title=card[constants.TRELLO_NAME], 
+                        listId=card[constants.TRELLO_IDLIST] )
+            self.items.insert(i, item)
```

Comment:
Why maintain self.items?

It adds complication to all of your functions. You could just build the list of items in `get_items`